2007-12-08  Ulrich Drepper  <drepper@redhat.com>

	[BZ #5428]
	* wcsmbs/wchar.h: Unconditionally undefine __need_mbstate and
	__need_wint_t.

Index: wcsmbs/wchar.h
===================================================================
RCS file: /cvs/glibc/libc/wcsmbs/wchar.h,v
retrieving revision 1.75
retrieving revision 1.76
diff -u -d -p -r1.75 -r1.76
--- wcsmbs/wchar.h	26 Nov 2007 22:50:59 -0000	1.75
+++ wcsmbs/wchar.h	10 Dec 2007 01:40:23 -0000	1.76
@@ -839,4 +839,9 @@ __END_DECLS
 
 #endif	/* _WCHAR_H defined */
 
+/* Undefined all __need_* constants in case we are included to get those
+   constants but the whole file was already read.  */
+#undef __need_mbstate_t
+#undef __need_wint_t
+
 #endif /* wchar.h  */
